R/utility.R

```diff
@@ -179,3 +179,28 @@ getJobResult <- function(jobId = "", ...){
 
   return(results)
 }
+
+validateClusterConfig <- function(clusterFilePath){
+  if (file.exists(clusterFilePath)) {
+    pool <- rjson::fromJSON(file = clusterFilePath)
+  }
+  else{
+    pool <- rjson::fromJSON(file = file.path(getwd(), clusterFilePath))
+  }
+
+  stopifnot(!is.null(pool$poolSize))
+  stopifnot(!is.null(pool$poolSize$dedicatedNodes))
+  stopifnot(!is.null(pool$poolSize$lowPriorityNodes))
+
+  stopifnot(is.character(pool$name))
+  stopifnot(is.character(pool$vmSize))
+  stopifnot(is.character(pool$poolSize$autoscaleFormula))
+
+  stopifnot(is.double(pool$poolSize$dedicatedNodes$min))
+  stopifnot(is.double(pool$poolSize$dedicatedNodes$max))
+  stopifnot(is.double(pool$poolSize$lowPriorityNodes$min))
+  stopifnot(is.double(pool$poolSize$lowPriorityNodes$max))
+  stopifnot(is.double(pool$maxTasksPerNode))
+
+  TRUE
```

Comment:
Additional validations:
- Make sure there is at least an entry for low pri or dedicated (or both if that is easier)

I don't think we'll need to validate all the possible options since the service will reject the pool creation call for bad values (e.g. bad vm size name) but we should make sure all 'ux improvements' we provide are tested for.